Add structured logging with structlog

markbreedlove opened the pull request:
#### What are the relevant tickets?

https://trello.com/c/ngDv49Sm/79-investigate-structlog-for-cleaning-logs-improving-observability

#### What's this PR do?

It introduces [structlog](https://www.structlog.org/en/stable/index.html), which wraps `logging` to provide some formatting conveniences. The most important improvement is creating structured, JSON-formatted messages that are easier for our log aggregators to parse, for us to monitor with Elastalert, and for us to analyze in Kibana.

I took the liberty in commit 4b74272 of changing some of the log messages. This commit can be backed out if this is too much at the moment.

#### How should this be manually tested?

If you have the `DEBUG` environment variable set, you should observe nicer human-readable log messages. (If you optionally install the `colorama` package, they will be color-coded.) Without the `DEBUG` flag, the messages will be JSON-encoded with UNIX timestamps.

mbertrand commented on cloudsync/tasks.py:
```diff
@@ -70,7 +70,7 @@ def stream_to_s3(self, video_id):
         video = Video.objects.get(id=video_id)
     except (Video.DoesNotExist, Video.MultipleObjectsReturned):
-        log.error("Exception retrieving video with id %d", video_id)
+        log.error("Exception retrieving video", video_id=video_id)
```
One issue with the new approach is that a separate sentry error seems to be created for every distinct video_id.

For example, testing locally created these 2 separate errors in sentry:
https://sentry.io/organizations/mit-office-of-digital-learning/issues/1588586740/?project=194353
https://sentry.io/organizations/mit-office-of-digital-learning/issues/1588586642/?project=194353

in addition to creating this issue, with the 2 lumped together:
https://sentry.io/organizations/mit-office-of-digital-learning/issues/1588573250/events/?project=194353


whereas the current code just lumps them under the same error:
https://sentry.io/organizations/mit-office-of-digital-learning/issues/1588573222/events/?project=194353

Is there a way to configure this to avoid that?

blarghmatey replied:
There should be an option in the Sentry configuration to tell it how to aggregate the errors. Might be worth looking at https://github.com/kiwicom/structlog-sentry

markbreedlove replied:
I'll have a look at the Sentry settings.

The funny thing is that I thought this would deduplicate Sentry issues.

markbreedlove replied:
I've just added the `structlog-sentry` module that @blarghmatey mentioned above. It's using the `SentryJsonProcessor`, which is supposed to prevent duplicate events, per https://github.com/kiwicom/structlog-sentry#logging-as-json ... @mbertrand, can you trigger another error so we can see what it looks like now in Sentry?

markbreedlove commented:
Turns out there are some test failures related to a few things that are my responsibility, so I'm working on those now ...

Thanks for the feedback so far. I'll address it when I've dealt with the test issues on my end.

markbreedlove commented:
I've just rebased and resolved various conflicts. In addition, I have squashed various commits that were trying to resolve pylint issues.

I'm going to look further now into deduplicating Sentry issues, mentioned [above](https://github.com/mitodl/odl-video-service/pull/865#discussion_r400416286).

mbertrand commented:
@markbreedlove Here's the latest errors, looks like they are being grouped together now:
https://sentry.io/organizations/mit-office-of-digital-learning/issues/1588573222/events/?project=194353
https://sentry.io/organizations/mit-office-of-digital-learning/issues/1588573250/events/?project=194353

mbertrand commented:
@markbreedlove sorry, disregard the above message, I hadn't restart the containers after switching branches.  Still getting one sentry error per video id:
https://sentry.io/organizations/mit-office-of-digital-learning/issues/1595777401/events/?project=194353
https://sentry.io/organizations/mit-office-of-digital-learning/issues/1595777339/events/?project=194353

markbreedlove commented:
Hey, @mbertrand :  Since it's been a few days, I was wondering if you could try that error event again in your dev environment. I want to be certain that this is on my `markbreedlove/structlog` branch. If it's still sending the `message` formatted as it is, it seems to go against what the `structlog-sentry` documentation [advertises](https://github.com/kiwicom/structlog-sentry#logging-as-json).

I can't tell from the Sentry error page what commit was deployed.

mbertrand commented:
Rebuilt all my containers, latest commit on this branch that I have is `daed22a443291c166bc41e777c7095c15c8fdd18` on 3/31:

Still getting separate errors here:
https://sentry.io/organizations/mit-office-of-digital-learning/issues/1601612935/?project=194353&query=is%3Aunresolved
https://sentry.io/organizations/mit-office-of-digital-learning/issues/1601612853/?project=194353&query=is%3Aunresolved

But they are grouped together here:
https://sentry.io/organizations/mit-office-of-digital-learning/issues/1588573250/events/?project=194353

markbreedlove commented:
Could you possibly try again? I have changed the configuration to exactly match the example given at https://github.com/kiwicom/structlog-sentry#logging-as-json where it's advertised as preventing duplicate entries.

markbreedlove commented:
@mbertrand : I've made another commit that changes the settings a little to match what I've seen work in my dev environment. Sentry is recording multiple events under the same issue.
Look in Sentry for `BadExceptionThree` to see what I mean. That's just with some test code. I am eager to see if it works as expected with your OVS container.

The new settings also [fine-tune the tags](https://github.com/mitodl/odl-video-service/pull/865/commits/331b1bb4caecb9cab1aef7cbb57a465b9573ac0d#diff-84aee3f4dade8d538db6e60259cdf0ccR27) that are applied to the Sentry issue. I hope that they will be useful.

mbertrand commented:
Still getting separate errors.  The tag keys (in this case, `video_id`) seem to be the culprit.  Did you use a tag key with the BadExceptionThree errors?

https://sentry.io/organizations/mit-office-of-digital-learning/issues/1603446992/?project=194353&query=is%3Aunresolved
https://sentry.io/organizations/mit-office-of-digital-learning/issues/1603446927/?project=194353&query=is%3Aunresolved

markbreedlove commented:
Yes, there have been tag keys for environment and level, and I have just generated another message with an additional tag key, `video_id`. The events keep getting rolled up successfully into the existing error in Sentry.

I'll keep investigating.

markbreedlove commented:
It's like Sentry is not parsing the message properly in the case of your actual dev environment. My test case has little complexity, and it's easy to rely on good defaults having been set. I wonder if, somehow, the mime type of the message to Sentry in your environment is not being set -- or some such thing -- that prevents the log message from being parsed as JSON. Sentry is assigning the literal string of the log message as the title of the error, for instance. In my case, there is sufficient information in the API call to distinguish the class of the Exception (`BadExceptionThree`) versus the string of the message.

markbreedlove commented:
I have an idea. It seems there's no point in logging something to Sentry without an exception trace. The `log()` call that I had generating your Sentry message was doing so without the `exc_info=True` parameter. It could be that was causing it to fall back to the whole JSON string.

I've just made a commit. Can you try with that?

If that _does_ work then I need to check the code for any `log.error()` calls that are made without `exc_info=True`.

markbreedlove commented:
Sorry, it looks like the `exc_info` parameter may not be the issue. I put a `log.error()` call in my test code without `exc_info=True` and it was parsed correctly by Sentry.

I'll do some more experiments on my end.

markbreedlove commented:
@mbertrand : I'm wondering what you think about this.  My test code is using the `sentry-sdk` package (the "sentry.python SDK"), but OVS is using `raven-python`. There are warnings on the Sentry error pages that recommend either upgrading `raven-python` to 6.10.0, or migrating to the sentry.python SDK. How easy would that be to change,  or try out in your dev environment?

markbreedlove commented:
[Sentry's Switching to Sentry-Python guide](https://forum.sentry.io/t/switching-to-sentry-python/4733)

If you have other things to do, I could probably take this on today.

mbertrand commented:
Nathan made a [PR to do just that for open-discussions](https://github.com/mitodl/open-discussions/pull/2586/files) a few months back, seems like it's worth doing for OVS too, especially because there was some kind of recent video performance issue seemingly caused by the raven JS code in Linux.  @pdpinch what do you think?

markbreedlove commented:
Heh heh, I should not have been so ambitious ... I'm grepping through the code now and it appears that somebody else could do that upgrade from Raven faster than I could.

markbreedlove commented:
Now that issue #868 is closed I will rebase this branch.

markbreedlove commented:
Hey, @mbertrand, I've done a rebase to bring in your sentry-sdk update, but the Travis build is failing:
https://travis-ci.com/github/mitodl/odl-video-service/jobs/319368394#L1071

I'm having a hard time identifying the change that's causing this, unless it's one of the dependencies in requirements.txt that I updated when I had to run `pip-compile`.

Can you provide any insight into this?
Thanks!

mbertrand commented:
@markbreedlove that might be a flaky test that only fails occasionally.  It should be fixed regardless, but I wonder if it fails consistently if you restart the travis build?  As far as I know that code hasn't been touched in awhile.

markbreedlove commented:
Still getting the error.

I'm not able to run `ui/signals_test.py` locally yet; it would appear because I don't have my environment all set up.

mbertrand commented:
ok I'll download this branch and figure out what's wrong with that test

markbreedlove commented:
Thanks for your help, @mbertrand ! The CI tests pass now.

mbertrand commented:
Looks like it worked! https://sentry.io/organizations/mit-office-of-digital-learning/issues/1609611188/?project=194353&query=is%3Aunresolved

mbertrand commented on cloudsync/tasks.py:
```diff
@@ -236,11 +243,17 @@ def upload_youtube_videos():
                 break
         except:  # pylint: disable=bare-except
-            log.exception("Error uploading video %s to Youtube: %s", video.hexkey, youtube_video.status)
+            log.error("Error uploading videoto Youtube",
```
minor typo: "videoto" -> "video to"

mbertrand commented on odl_video/logging.py:
```diff
@@ -0,0 +1,47 @@
+        structlog.stdlib.filter_by_level,
+        structlog.processors.TimeStamper(),
+        SentryJsonProcessor(level=logging.ERROR,
+                            tag_keys=['environment', 'level', 'logger',
```
does there need to be a tag_key for every variable used in an error message?  For instance, [this code](https://github.com/mitodl/odl-video-service/pull/865/files#diff-0da2c698f19bf424e9fdcd1e851eda69R255) adds `status` to the error message but it isn't showing up in sentry here: https://sentry.io/organizations/mit-office-of-digital-learning/issues/1615059130/?project=194353&query=is%3Aunresolved

markbreedlove replied:
That `tag_keys` List in logging.py is a whitelist of tags that it will set in the Sentry issue that come from properties set in the `log.error()` call. The `status` tag was not set in the Sentry error above because it wasn't on that list, despite the code having executed `raise HttpAccessTokenRefreshError(error_msg, status=resp.status)`. I can add `status` to `tag_keys`.

If you feel that you're seeing too many tags in the errors I can recduce them. I'm noticing now that `structlog` or `sentry-sdk` is adding tags on its own. Is that OK? For instance, I didn't configure anything to make it add `trace` or `transaction`. Are you OK with the proliferation of tags?

mbertrand replied:
I'm fine with the extra tags that structlog adds on its own.

For things like `status`, can it somehow be included in the message without adding a tag for it?  I can foresee situations where I'd want to add some extra info to a particular error message without wanting to add it as a tag to the structlog setup.

For example, here is the error on CI using the old logging setup, the status shows up as variable #1 in the message: https://sentry.io/organizations/mit-office-of-digital-learning/issues/1615059130/events/c883c45d08484cec9f9cca94c89d555e/?project=194353

But on my dev instance it is not visible anywhere as far as I can tell:
https://sentry.io/organizations/mit-office-of-digital-learning/issues/1615059130/events/9b1aa3456c44436a8666715b10d2a435/?project=194353

markbreedlove replied:
Sure, I'm happy to put the video ID back into the Message string of the log message, and to put variable values back into messages where you would like them.  I'll do this one now.

The cool thing about the new error (the 2nd link above) is there's a tag now named `video_hexkey` that has the video ID.  Unlike before, you can click on that tag and see all errors in the system pertaining to that particular video, instead of having to do a text search.

markbreedlove replied:
(With structured logging, a similar thing will happen in Kibana. There will automagically be a field that gets indexed in Elasticsearch for that video ID. You'll be able to click on a filter icon in the record you're looking at to filter on that video ID, instead of having to go up to the search bar and search on it.)

markbreedlove replied:
Changed in commit ec20c21

mbertrand replied:
I agree it's useful to have a video hexkey tag, maybe a video status tag too, but occasionally we might want to have some extra value in one particular error message that wouldn't be worth adding a tag for, so still having the option of just including it as a formatted string would be good, as in the code comment below `"Error uploading video to Youtube: %s", positional_args=(youtube_video.status,)`

mbertrand commented on odl_video/logging.py:
```diff
@@ -0,0 +1,47 @@
+
+structlog.configure(
+    processors=structlog_processors_per_debug[settings.DEBUG],
+    logger_factory=structlog.stdlib.LoggerFactory()
```
if you add 
```python
wrapper_class=structlog.stdlib.BoundLogger,
```

then you won't have to replace occurrences of `log.exception('oops')` in master with `log.error('oops', exc_info=True)` in this branch.

markbreedlove replied:
OK, changed in commit f9feae2

markbreedlove commented on odl_video/logging.py:
```diff
@@ -0,0 +1,47 @@
+        structlog.dev.set_exc_info,
+        structlog.processors.format_exc_info,
+        structlog.processors.TimeStamper(fmt='iso'),
+        structlog.dev.ConsoleRenderer()
```
OK.

Commit: f74483b

markbreedlove commented on cloudsync/tasks.py:
```diff
@@ -191,7 +192,7 @@ def schedule_retranscodes(self):
     except:  # pylint: disable=bare-except
         error = "schedule_retranscodes threw an error"
-        log.exception(error)
+        log.error(error)
```
I had to change `exception` to `error` in commit 6ccbfb6fbd5ea2a7810d5e0101f5667a5c7a8903 because `structlog-sentry` raises an `AttributeError` when it tries to use `getattr()` to get `logging.Logger`'s log method. (See commit comment.)

mbertrand replied:
I think adding `wrapper_class=structlog.stdlib.BoundLogger,` fixes that problem?

markbreedlove replied:
I'll change it and see. Does `exception()` do anything that `error()` with a stacktrace doesn't do? I had checked the `logging` documentation and it seemed to say it was just `error()` with a stacktrace.

mbertrand replied:
log.exception should include the stacktrace automatically, so you don't have to include `exc_info=True`)

markbreedlove replied:
Changed at https://github.com/mitodl/odl-video-service/pull/865/commits/dca212eeda1dd02a88bff98cba93f18493a1f16b#diff-0da2c698f19bf424e9fdcd1e851eda69R195

markbreedlove replied:
Right, the fact that `error()` doesn't include the stacktrace by default is why I had `exc_info=True` in there. I had switched to it because I was getting an AttributeError with `exception()`. But I think that besides adding a stacktrace `exception()` isn't any different than `error()`. If the currently-running CI builds succeed then we're fine. If we keep getting AttributeError I'll need to change back to `error()`. Fortunately, the exception will always be passed to Sentry with BoundLogger, but BoundLogger only affects Sentry's behavior. It won't make sure that the exception gets into the log message that gets shipped to ELK.

markbreedlove commented on cloudsync/api.py:
```diff
@@ -316,7 +318,8 @@ def parse_lecture_video_filename(filename):
     matches = re.search(rx, filename)
     if not matches or len(matches.groups()) != 5:
-        log.exception('No matches found for filename %s with regex %s', filename, rx)
+        log.error('No matches found',
```
Changed at https://github.com/mitodl/odl-video-service/pull/865/commits/dca212eeda1dd02a88bff98cba93f18493a1f16b#diff-cb84ab42f8fed63aa4dcf2881ec1f47eR321-R323

markbreedlove commented on cloudsync/api.py:
```diff
@@ -385,7 +390,8 @@ def upload_subtitle_to_s3(caption_data, file_data):
         except ClientError:
-            log.exception('Could not delete old subtitle from S3: %s', vt.s3_object_key)
+            log.error('Could not delete old subtitle from S3',
+                      s3_object_key=vt.s3_object_key)
```
Changed at https://github.com/mitodl/odl-video-service/pull/865/commits/dca212eeda1dd02a88bff98cba93f18493a1f16b#diff-cb84ab42f8fed63aa4dcf2881ec1f47eR394-R396

markbreedlove commented on cloudsync/tasks.py:
```diff
@@ -211,11 +212,14 @@ def update_video_statuses(self):
         except EncodeJob.DoesNotExist:
             # Log the exception but don't raise it so other videos can be checked.
-            log.exception("No EncodeJob object exists for video id %d", video.id)
+            log.error("No EncodeJob object exists",
```
Changed at https://github.com/mitodl/odl-video-service/pull/865/commits/dca212eeda1dd02a88bff98cba93f18493a1f16b#diff-0da2c698f19bf424e9fdcd1e851eda69R215-R217

markbreedlove commented on cloudsync/tasks.py:
```diff
@@ -211,11 +212,14 @@ def update_video_statuses(self):
-            log.exception("AWS error when refreshing job status for video %d: %s", video.id, exc.response)
+            log.error("AWS error when refreshing job status",
+                      video_id=video.id,
+                      response=exc.response)
```
Changed at https://github.com/mitodl/odl-video-service/pull/865/commits/dca212eeda1dd02a88bff98cba93f18493a1f16b#diff-0da2c698f19bf424e9fdcd1e851eda69R221-R225

markbreedlove commented on cloudsync/tasks.py:
```diff
@@ -236,11 +240,17 @@ def upload_youtube_videos():
             youtube_video.save()
         except HttpError as error:
-            log.exception("HttpError uploading video %s to Youtube: %s", video.hexkey, youtube_video.status)
+            log.error("HttpError uploading video %s to Youtube: %s",
```
Changed at https://github.com/mitodl/odl-video-service/pull/865/commits/dca212eeda1dd02a88bff98cba93f18493a1f16b#diff-0da2c698f19bf424e9fdcd1e851eda69R246-R249

markbreedlove commented on cloudsync/tasks.py:
```diff
@@ -236,11 +240,17 @@ def upload_youtube_videos():
                 break
         except:  # pylint: disable=bare-except
-            log.exception("Error uploading video %s to Youtube: %s", video.hexkey, youtube_video.status)
+            log.error("Error uploading video %s to Youtube: %s",
```
Changed at https://github.com/mitodl/odl-video-service/pull/865/commits/dca212eeda1dd02a88bff98cba93f18493a1f16b#diff-0da2c698f19bf424e9fdcd1e851eda69R253-R256

markbreedlove commented on cloudsync/tasks.py:
```diff
@@ -302,7 +313,9 @@ def update_youtube_statuses(self):
             yt_video.status = YouTubeStatus.FAILED
             yt_video.save()
-            log.exception('Status of YoutubeVideo %s for Video %s not found.', yt_video.id, yt_video.video_id)
+            log.error('Status of YoutubeVideo not found.',
```
Changed at https://github.com/mitodl/odl-video-service/pull/865/commits/dca212eeda1dd02a88bff98cba93f18493a1f16b#diff-0da2c698f19bf424e9fdcd1e851eda69R319-R321

markbreedlove commented on cloudsync/tasks.py:
```diff
@@ -322,10 +335,13 @@ def monitor_watch_bucket(self):
         except ClientError as exc:
             # Log ClientError, raise later so other files can be processed.
-            log.exception("AWS error when ingesting file from watch bucket %s: %s", key.key, exc.response)
+            log.error("AWS error when ingesting file from watch bucket",
```
Changed at https://github.com/mitodl/odl-video-service/pull/865/commits/dca212eeda1dd02a88bff98cba93f18493a1f16b#diff-0da2c698f19bf424e9fdcd1e851eda69R341-R343

markbreedlove commented on cloudsync/tasks.py:
```diff
@@ -322,10 +335,13 @@ def monitor_watch_bucket(self):
         except Exception as exc:  # pylint: disable=broad-except
             # Log any other exception, raise later so other files can be processed.
-            log.exception("AWS error when ingesting file from watch bucket %s: %s", key.key, str(exc))
+            log.error("AWS error when ingesting file from watch bucket",
```
Changed at https://github.com/mitodl/odl-video-service/pull/865/commits/dca212eeda1dd02a88bff98cba93f18493a1f16b#diff-0da2c698f19bf424e9fdcd1e851eda69R346-R347

markbreedlove commented on odl_video/logging.py:
```diff
@@ -0,0 +1,51 @@
+                                      'runtime', 'server_name', 'video_id',
+                                      'video_hexkey', 's3_object_key',
+                                      'filename', 'youtubevideo_id',
+                                      'youtubevideo_video_id'],
```
Added: https://github.com/mitodl/odl-video-service/pull/865/commits/dca212eeda1dd02a88bff98cba93f18493a1f16b#diff-84aee3f4dade8d538db6e60259cdf0ccR34

markbreedlove commented on mail/tasks.py:
```diff
@@ -97,7 +99,9 @@ def send_notification_email(video):
             _send_debug_email(video=video, email_kwargs=email_kwargs)
     except:  # pylint: disable=bare-except
-        log.exception('Impossible to send notification for video %s with status %s', video.hexkey, video.status)
+        log.error('Impossible to send notification',
```
Changed at https://github.com/mitodl/odl-video-service/pull/865/commits/b1ac2d73f322d2b5dbc649fcb7323cf473a7a9eb#diff-7676b50b98db5916c53dca8f267f0aa4R102-R104

markbreedlove commented on techtv2ovs/tasks.py:
```diff
@@ -71,7 +71,9 @@ def process_videofiles(self, ttv_id, files, run_copy):
             )
         except Exception as exc:
-            log.exception("Error processing video file %s for ttv video %s", s3file['Key'], ttv_video.id)
+            log.error("Error processing video file for ttv video",
```
Changed at https://github.com/mitodl/odl-video-service/pull/865/commits/b1ac2d73f322d2b5dbc649fcb7323cf473a7a9eb#diff-e181fefa72558fdff4957f01711f74beR74-R76

mbertrand commented on cloudsync/api.py:
```diff
@@ -316,7 +318,9 @@ def parse_lecture_video_filename(filename):
-        log.exception('No matches found for filename %s with regex %s', filename, rx)
+        log.exception('No matches found for filename %s with regex %s',
+                      positional_args=(filename, rx),
+                      filename=filename, regex=rx)
```
I think this can now be shortened to:
```python
log.exception('No matches found for filename %s with regex %s',
                      positional_args=(filename, rx))
```
because `filename` and `regex` aren't in the tags

markbreedlove replied:
I'll make a commit to add `filename` to the tags, because I could see it being useful, but I'll remove `rx` from the arguments to the `exception()` call because I don't anticipate anyone searching on that.

markbreedlove replied:
`filename` is in the tags [here](https://github.com/mitodl/odl-video-service/blob/6f941e1a3bf77e4efa5b261e7554a5e7e2ce5e32/odl_video/logging.py#L33) so I'll just get rid of `regex=rx` in the function args.

markbreedlove replied:
Changed in 17e0a9e

mbertrand commented on cloudsync/api.py:
```diff
@@ -385,7 +391,9 @@ def upload_subtitle_to_s3(caption_data, file_data):
-            log.exception('Could not delete old subtitle from S3: %s', vt.s3_object_key)
+            log.exception('Could not delete old subtitle from S3: %s',
+                          positional_args=(vt.s3_object_key,),
+                          s3_object_key=vt.s3_object_key)
```
`s3_object_key` is a tag so no need for positional_args here:

```python
log.exception('Could not delete old subtitle from S3',
                          s3_object_key=vt.s3_object_key)
```

markbreedlove replied:
Changed in 17e0a9e

mbertrand commented on cloudsync/tasks.py:
```diff
@@ -211,11 +212,17 @@ def update_video_statuses(self):
-            log.exception("No EncodeJob object exists for video id %d", video.id)
+            log.exception("No EncodeJob object exists for video id %d",
+                          positional_args=(video.id,),
+                          video_id=video.id)
```
Since `video_id` is a tag, no need for positional_args:
```python
log.exception("No EncodeJob object exists for video id",
                          video_id=video.id)
```

markbreedlove replied:
Changed in 17e0a9e

mbertrand commented on cloudsync/tasks.py:
```diff
@@ -211,11 +212,17 @@ def update_video_statuses(self):
+                          "video %d: %s",
+                          positional_args=(video.id, exc.response),
+                          video_id=video.id,
+                          response=exc.response)
```
```python
log.exception("AWS error when refreshing job status", video_id=video.id)
```

markbreedlove replied:
I'll take out the positional_args, but I'll keep the response as a structured log field so that we can have it in Kibana.

markbreedlove replied:
Changed in 17e0a9e

mbertrand commented on cloudsync/tasks.py:
```diff
@@ -236,11 +243,17 @@ def upload_youtube_videos():
+            log.exception("HttpError uploading video %s to Youtube: %s",
+                          positional_args=(video.hexkey, youtube_video.status),
+                          video_hexkey=video.hexkey,
+                          status=youtube_video.status)
```
```python
log.exception("HttpError uploading video to Youtube",
                          video_hexkey=video.hexkey,
                          video_status=youtube_video.status)
```

markbreedlove replied:
Changed in 17e0a9e

mbertrand commented on cloudsync/tasks.py:
```diff
@@ -236,11 +243,17 @@ def upload_youtube_videos():
+            log.exception("Error uploading video %s to Youtube: %s",
+                          positional_args=(video.hexkey, youtube_video.status),
+                          video_hexkey=video.hexkey,
+                          status=youtube_video.status)
```
```python
log.exception("Error uploading video to Youtube",
                          video_hexkey=video.hexkey,
                          video_status=youtube_video.status)
```

markbreedlove replied:
Changed in 17e0a9e
